app/controllers/tasks_controller.rb

```diff
@@ -1,8 +1,73 @@
 class TasksController < ApplicationController
+
   def index
+    @filter = params[:category]
+    @all_categories = %w[Inventory Register Engineering]
+    @tasks = if @filter.blank?
+               Task.all
+             else
+               Task.where(category: @filter)
+             end
+    incomplete_task = @tasks.where(completed: false).sort_by { |task| [-task.priority, task.added] }
+    completed_tasks = @tasks.where(completed: true).sort_by { |task| [task.complete_time] }.reverse
+    @tasks = incomplete_task + completed_tasks
+    @num_incomplete = @tasks.select { |task| task.completed == false }.count
```

Comment:
there's a lot of redundant code here.  if you already have `incomplete_task`, why isn't `@num_incomplete` just the size of that collection?  this code could use some tightening.